src/convertToObject.js

```diff
@@ -3,16 +3,29 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example
+ * in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const propertiesObject = {};
+
+  sourceString
+    .split(';')
+    .map(line => line.split(':').map(element => element.trim()))
+    .forEach(propertyPair => {
+      if (propertyPair[0]) {
```

Comment:
You are checking if propertyPair[0] exists before adding it to the propertiesObject. This will add undefined as a key if propertyPair[1] is not defined. Consider checking both propertyPair[0] and propertyPair[1] before adding them to the object to prevent this.